pkg/cmd/server/kubernetes/master.go

```diff
@@ -171,5 +172,13 @@ func (c *MasterConfig) createSchedulerConfig() (*scheduler.Config, error) {
 	}
 
 	// if the config file isn't provided, use the default provider
-	return configFactory.CreateFromProvider(factory.DefaultProvider)
+	provider, err := factory.GetAlgorithmProvider(factory.DefaultProvider)
+	if err != nil {
+		return nil, err
+	}
+	// Register origin specific predicates and priority functions.
+	// TODO: Move generic predicates in origin to upstream k8s.
+	provider.FitPredicateKeys.Insert(originscheduler.RegisterOriginPredicates()...)
```

Comment:
Also, you should not insert originscheduler.RegisterOriginPredicates() anyways, since that method is expected to register (over time) multiple predicates and not all of them would be included by default.